### ISSUE TITLE
Confusing (or incorrect?) parse result with `try_table`

### ISSUE THREAD

**Issue by osa1** [bytecodealliance/wasm-tools#1340]
When I parse the following .wast:

```wast
(module
  (tag $e-i32-i32 (param i32 i32))

  (func $throw-1-2 (i32.const 1) (i32.const 2) (throw $e-i32-i32))

  (func (export "test-throw-1-2")
    (block $h (result i32 i32)
      (try_table (catch $e-i32-i32 $h) (call $throw-1-2))
      (return))
    (if (i32.ne (i32.const 2)) (then (unreachable)))
    (if (i32.ne (i32.const 1)) (then (unreachable)))))
```

With the main branch, I get this parse tree: (simplified by removing some unrelated parts)

```
Module {
    ...
    kind: Text(
        [
            ...,
            Func(
                Func {
                    ...
                    kind: Inline {
                        locals: [],
                        expression: Expression {
                            instrs: [
                                Block(...),
                                Call(Id("throw-1-2")),
                                TryTable(
                                    TryTable {
                                        block: BlockType { ... },
                                        catches: [TryTableCatch { kind: Catch(Id("e-i32-i32")), label: Id("h")  }],
                                    },
                                ),
                                Return,
                                End(None), // end TryTable
                                I32Const(2),
                                I32Ne,
                                If(...),
                                Unreachable,
                                End(None), // end If
                                I32Const(1),
                                I32Ne,
                                If(...),
                                Unreachable,
                                End(None), // end If
                            ],
                        },
                    },
                    ty: TypeUse {
                        index: None,
                        inline: None,
                    },
                },
            ),
        ],
    ),
}
```

Here we have a `block`, `try_table`, and two `if`s. In total this should have 4 `end`s, but it only has 3.

Is this a bug, or is my reading is wrong?

<details>

<summary>Full debug output for the module:</summary>

```
[crates/spec-test/src/spec.rs:232] m = Module {
    span: Span {
        offset: 29,
    },
    id: None,
    name: None,
    kind: Text(
        [
            Tag(
                Tag {
                    span: Span {
                        offset: 39,
                    },
                    id: Some(
                        "e-i32-i32",
                    ),
                    name: None,
                    exports: InlineExport {
                        names: [],
                    },
                    ty: Exception(
                        TypeUse {
                            index: None,
                            inline: Some(
                                FunctionType {
                                    params: [
                                        (
                                            None,
                                            None,
                                            I32,
                                        ),
                                        (
                                            None,
                                            None,
                                            I32,
                                        ),
                                    ],
                                    results: [],
                                },
                            ),
                        },
                    ),
                    kind: Inline,
                },
            ),
            Func(
                Func {
                    span: Span {
                        offset: 75,
                    },
                    id: Some(
                        "throw-1-2",
                    ),
                    name: None,
                    exports: InlineExport {
                        names: [],
                    },
                    kind: Inline {
                        locals: [],
                        expression: Expression {
                            instrs: [
                                I32Const(
                                    1,
                                ),
                                I32Const(
                                    2,
                                ),
                                Throw(
                                    Id(
                                        "e-i32-i32",
                                    ),
                                ),
                            ],
                        },
                    },
                    ty: TypeUse {
                        index: None,
                        inline: None,
                    },
                },
            ),
            Func(
                Func {
                    span: Span {
                        offset: 143,
                    },
                    id: None,
                    name: None,
                    exports: InlineExport {
                        names: [
                            "test-throw-1-2",
                        ],
                    },
                    kind: Inline {
                        locals: [],
                        expression: Expression {
                            instrs: [
                                Block(
                                    BlockType {
                                        label: Some(
                                            "h",
                                        ),
                                        label_name: None,
                                        ty: TypeUse {
                                            index: None,
                                            inline: Some(
                                                FunctionType {
                                                    params: [],
                                                    results: [
                                                        I32,
                                                        I32,
                                                    ],
                                                },
                                            ),
                                        },
                                    },
                                ),
                                Call(
                                    Id(
                                        "throw-1-2",
                                    ),
                                ),
                                TryTable(
                                    TryTable {
                                        block: BlockType {
                                            label: None,
                                            label_name: None,
                                            ty: TypeUse {
                                                index: None,
                                                inline: None,
                                            },
                                        },
                                        catches: [
                                            TryTableCatch {
                                                kind: Catch(
                                                    Id(
                                                        "e-i32-i32",
                                                    ),
                                                ),
                                                label: Id(
                                                    "h",
                                                ),
                                            },
                                        ],
                                    },
                                ),
                                Return,
                                End(
                                    None,
                                ),
                                I32Const(
                                    2,
                                ),
                                I32Ne,
                                If(
                                    BlockType {
                                        label: None,
                                        label_name: None,
                                        ty: TypeUse {
                                            index: None,
                                            inline: None,
                                        },
                                    },
                                ),
                                Unreachable,
                                End(
                                    None,
                                ),
                                I32Const(
                                    1,
                                ),
                                I32Ne,
                                If(
                                    BlockType {
                                        label: None,
                                        label_name: None,
                                        ty: TypeUse {
                                            index: None,
                                            inline: None,
                                        },
                                    },
                                ),
                                Unreachable,
                                End(
                                    None,
                                ),
                            ],
                        },
                    },
                    ty: TypeUse {
                        index: None,
                        inline: None,
                    },
                },
            ),
        ],
    ),
}
```

</details>

**Comment by alexcrichton** [comment]
Thanks for the report! I believe the parsing of the s-expression form had an accidental bug which will be fixed by https://github.com/bytecodealliance/wasm-tools/pull/1333

**Comment by osa1** [comment]
This is fixed in version 70.0.0.
